fe/src/main/java/org/apache/doris/analysis/DateLiteral.java

```diff
@@ -319,7 +319,17 @@ protected void toThrift(TExprNode msg) {
 
     @Override
     protected Expr uncheckedCastTo(Type targetType) throws AnalysisException {
-        if (targetType.equals(this.type)) {
+        if (targetType.isDateType()) {
```

Comment:
If isDateType is true. The type could only be DATE and DATETIME.

So one `if  else` if enough?